# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../pdksh/initrd-usability.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

As we use the pdksh as the default shell in the initrd, it was a common
request to make it a bit more intuitive by default: history and editing +
tab completion - as most people are used from daily work already.

  - Rene Rebe <rene@exactcode.de>

--- pdksh-5.2.14/main.c	1999-05-11 15:59:52.000000000 +0200
+++ pdksh-5.2.14-hackery/main.c	2007-02-26 15:59:50.110424557 +0100
@@ -201,6 +201,11 @@
 	change_flag(FPOSIX, OF_SPECIAL, 1);
 #endif /* POSIXLY_CORRECT */
 
+	/* Set edit mode to emacs by default, may be overridden. */
+#if defined(EDIT) && defined(EMACS)
+	change_flag(FEMACS, OF_SPECIAL, 1);
+#endif /* EDIT && EMACS */
+
 	/* import enviroment */
 	if (environ != NULL)
 		for (wp = environ; *wp != NULL; wp++)
--- pdksh-5.2.14/emacs.c	1999-07-07 21:43:43.000000000 +0200
+++ pdksh-5.2.14-hackery/emacs.c	2007-02-26 16:03:05.517560175 +0100
@@ -268,7 +268,7 @@
 #else
 	{ XFUNC_transpose,		0, CTRL('T') },
 #endif
-	{ XFUNC_complete,		1, CTRL('[') },
+	{ XFUNC_complete,		0,	'\t' },
         { XFUNC_comp_list,		1,	'='  },
 	{ XFUNC_enumerate,		1,	'?'  },
         { XFUNC_expand,			1,	'*'  },
